### ISSUE TITLE
"Repeat while held" unchecks after restarting

### ISSUE THREAD

**Issue by kurisu-io** [Ryochan7/DS4Windows#2800]
**Describe the bug**
I have a macro assigned to my `Right Touch` trigger under special actions (a recorded macro). I have the "repeat while held" option enabled. It works perfectly fine. But after restarting my computer or closing out of DS4Windows and starting it up again, the "repeat while held" option becomes unchecked.

**To Reproduce**
1. Go edit a profile.
2. Under Special Actions, create or edit an action.
3. Record a macro and turn on repeat while held.
4. Save the special action and then save the profile.
5. Test that the macro repeats while the trigger is held.
6. Completely close out of DS4Windows and then restart it again.
7. Check the macro's options again and note that the "Repeat while held" option is unchecked.

**Expected behavior**
The "repeat while held" should have remained enabled even after computer restarts or DS4Windows restarts.

**Screenshots and Logs**
![image](https://user-images.githubusercontent.com/764773/215137333-70b69c28-88be-4ada-98ae-c60533acdf70.png)
The option in the red box unchecks after closing out of DS4Windows (or after rebooting the system) and starting DS4Windows again.

```
2023-01-28 01:06:37.5754|INFO|DS4Windows version 3.2.7
2023-01-28 01:06:37.5781|INFO|DS4Windows exe file: DS4Windows.exe
2023-01-28 01:06:37.5781|INFO|DS4Windows Assembly Architecture: x64
2023-01-28 01:06:37.5781|INFO|OS Version: Microsoft Windows NT 10.0.22621.0
2023-01-28 01:06:37.5781|INFO|OS Product Name: Windows 10 Pro
2023-01-28 01:06:37.5781|INFO|OS Release ID: 2009
2023-01-28 01:06:37.5781|INFO|System Architecture: x64
2023-01-28 01:06:37.5781|INFO|Logger created
2023-01-28 01:06:38.3241|INFO|Running as User
2023-01-28 01:06:38.3241|INFO|Some applications may block controller inputs. (Windows UAC Conflictions). Please go to https://docs.ds4windows.app/troubleshooting/kb-mouse-issues/#windows-not-responding-to-ds4ws-kb-m-commands-in-some-situations for more information and workarounds.
2023-01-28 01:06:38.3241|INFO|HidHide control device found
2023-01-28 01:06:39.3432|INFO|Starting...
2023-01-28 01:06:39.3432|INFO|Using output KB+M handler: SendInput
2023-01-28 01:06:39.3432|INFO|Connection to ViGEmBus 1.21.442.0 established
2023-01-28 01:06:39.3432|INFO|Searching for controllers...
2023-01-28 01:06:39.3432|INFO|Using Shared Mode
2023-01-28 01:06:39.3922|INFO|Found Controller: 88:03:4C:48:6D:07 (USB) (DualSense).
2023-01-28 01:06:39.5249|INFO|Plugging in virtual DS4 Controller in output slot #1
2023-01-28 01:06:39.5249|INFO|Associated input controller #1 (DualSense) to virtual DS4 Controller in output slot #1
2023-01-28 01:06:39.5422|INFO|Controller 1 is using Profile "Default". Battery level is 100%
```

**Desktop (please complete the following information):**
 - Controller Make and Model: Sony DualSense
 - OS: Windows 11 Pro
 - DS4Windows Version 3.2.7

**Comment by Ryochan7** [comment]
None of the extra settings were being saved for the Macro special action. Another core model problem. Change the flow of saving and this happens. Fixed in commit 9a3c842f897c20488a74c0a861a328e533979e94.

**Comment by kurisu-io** [comment]
Thank you for the fix. I updated to https://github.com/Ryochan7/DS4Windows/releases/tag/v3.2.8 and confirmed the settings are saving properly on my end. I'll close this now.
